user-defined-plugin.md

```diff
@@ -0,0 +1,261 @@
+# Add your Custom Data Sources with Plugin Framework
+
+Exploratory provides a framework with which developers can create their own data sources that can be invoked from Exploratory's Data Import Dialog.
+
+## Plugin Structure
+
+A User defined plugin consists of JSON format plugin definition meta file (`plugin.json`) and R script file (`plugin.r`).
+These two files need to be stored inside `~/.exploratory/plugins` directory as follows.
+
+`~/.exploratory/plugins/<plugin_name>/plugin.json`
+
+`~/.exploratory/plugins/<plugin_name>/scripts/plugin.r`
+ 
+
+## Create Plugin directory
+
+So let's say you want to create a plugin that uses `riem_measures` function from [riem](https://github.com/ropensci/riem) R package to get ASOS data via the Iowa Environment Mesonet. Then first you want to to create a directory called `riem_measures`. NOTE: The directory name is the plugin name and should be unique among user-defined plugins.
```

Comment:
"Then first you want to to create a directory called `riem_measures`."
This gave me an impression that the plugin name must be same as R package name.